src/Nethermind/Nethermind.JsonRpc/Modules/SingletonModulePool.cs

```diff
@@ -37,14 +40,14 @@ public SingletonModulePool(IRpcModuleFactory<T> factory, bool allowExclusive = t
             _allowExclusive = allowExclusive;
         }
         
-        public T GetModule(bool canBeShared)
+        public Task<T> GetModule(bool canBeShared)
```

Comment:
I guess there may be valid reasons but since most of the time we return the shared instance shall we use ValueTask instead?

Reply:
The shared instance is wrapped with `Task.FromResult` and uses task eliding already.